addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -541,6 +541,9 @@ MockServer.include({
                         ['res_id', '=', thread.id],
                         ['res_model', '=', thread_model],
             ]], {});
+            // search read returns many2one relations as an array [id, display_name].
+            // But the original route does not. Thus, we need to change it now.
+            followers.forEach(follower => follower.partner_id = follower.partner_id[0]);
```

Comment:
I'm not sure to understand why it has to be changed in this PR

Reply:
We used an integer field instead of the real `partner_id` field. Now that it is a `many2one` field, the search read method returns an array instead of a simple integer. On the server the `/mail/thread/data` route also returns an integer so we need to adapt it to match the server behavior.

Reply:
I see, well changing the field type to be correct could/should be done in a prep PR to avoid confusion here.